Kuadrant/kuadrant-operator
# local operator catalog raw file based format

### what

* Locally built custom OLM catalogs use (raw) File Based Catalog format instead of the deprecated SQLITE format.
From [SQLite-based catalogs](https://docs.openshift.com/container-platform/4.9/operators/admin/olm-managing-custom-catalogs.html#olm-managing-custom-catalogs-sqlite),

```
The SQLite database format for Operator catalogs is a deprecated feature. Deprecated functionality is still included in OpenShift Container Platform and continues to be supported; however, it will be removed in a future release of this product and is not recommended for new deployments.
```

* Update github actions to push file based catalog format images
* Makefile
  * target renamed: `deploy-olm` -> `deploy-catalog`
  * target renamed: `undeploy-olm` -> `undeploy-catalog`
  * function renamed: `go-get-tool` -> `go-install-tool`
  * `bundle-dependencies` target removed. Functionality moved to the `bundle` target.
  * replace `envsubst` operations to update yaml objects with external env variarbles with the `yq` tool
  * moved all catalog related logic to `make/catalog.mk`
  * New: `catalog` target to build catalog manifests
* Documentation oriented to contributors `doc/development.md`:
  * New section: `Deploy kuadrant operator using OLM`
  * New section: "Build custom OLM catalog`
  * Added Table of Contents.

TODO

- [x] github actions

### Verification steps

Build local catalog. It will include kuadrant operator, limitador operator and authorino operator

```
make catalog
make catalog-build CATALOG_IMG=quay.io/kuadrant/kuadrant-operator-catalog:local-testing
```

Push the image to Quay.io. Currently it is required because the catalogsource sets the image pull policy to `Always`, so even if the image is loaded in the kind cluster, it will fail if the image is not in quay.

```
make catalog-push CATALOG_IMG=quay.io/kuadrant/kuadrant-operator-catalog:local-testing
```

Create kind cluster

```
make kind-create-cluster
```

Deploy OLM

```
make install-olm
```

Deploy kuadrant operator using OLM

```
make deploy-catalog CATALOG_IMG=quay.io/kuadrant/kuadrant-operator-catalog:local-testing
```

Check the kuadrant operator, limitador operator and authorino operator are up and running

```
k get pods -n kuadrant-system
NAME                                                              READY   STATUS      RESTARTS   AGE
43539917290efc2e6f33b021e25da1f7a9f150d8d276acf92fd09b--1-wqggr   0/1     Completed   0          34s
6427bb5aac2edc4750e505a33be4eb4b56923d3aed4213abecdc70--1-v4lk5   0/1     Completed   0          34s
aecaf21d9bde4883921cefdd0ba3a142f079e897473c5d40fd8a1a--1-2249v   0/1     Completed   0          34s
authorino-operator-7d9968586c-x4kbl                               1/1     Running     0          16s
kuadrant-operator-catalog-rkmdn                                   1/1     Running     0          79s
kuadrant-operator-controller-manager-6688b8bfdf-m4rwh             1/1     Running     0          13s
limitador-operator-controller-manager-7f5dcdd976-66bcs            1/1     Running     0          11s
```
